RdlMigration/ConvertRDL.cs

```diff
@@ -560,6 +580,47 @@ private XElement CreateDataSourceNode(XDocument doc, DataSourceDefinition curren
             return currNode;
         }
 
+        /// <summary>
+        /// This method takes in an XDocument, looks for valid subreports within, and attempts 
+        /// to upload these subreports.
+        /// </summary>
+        /// <param name="doc"> The XDocument of local rdl File.</param>
+        private void DiscoverSubreports(XDocument doc)
+        {
+            var subreports = doc.Descendants(doc.Root.Name.Namespace + "Subreport");
+            var subreportPaths = new List<string>();
+
+            foreach (XElement subreport in subreports)
+            {
+                var subreportName = subreport.Elements().First().Value;
+                var subreportPath = Path.Combine(rootFolder, subreportName);
+                subreportPath = Path.GetFullPath(subreportPath).Replace("\\", "/");
+                subreportPath = subreportPath.Substring(subreportPath.IndexOf('/'));   // file path from server root
+                subreportName = Path.GetFileName(subreportPath);                       // clean file name with no folder path
+                
+                if (!rdlFileIO.IsFile(subreportPath))
+                {
+                    Trace($"SUBREPORT FAIL : {subreportPath} Subreport does not exist");
+                    continue;
+                }
+
+                if (reportNameMap.TryAdd(subreportName, subreportPath))
+                {
+                    subreportPaths.Add(subreportPath);
+                    Trace($"SUBREPORT : Attempting to upload subreport from {subreportPath}");
+                }
+                else
+                {
+                    Trace($"SUBREPORT : {subreportPath} A report with the same file name is already in the upload queue");
```

Comment:
>Trace($"SUBREPORT : {subreportPath} A report with the same file name is already in the upload queue"); [](start = 20, length = 102)

this message doesn't seem helpful to the user.

Reply:
edited to be more clear and consistent with current naming conflict message.